### ISSUE TITLE
runtime: goroutine stack exceeds 1000000000-byte limit

### ISSUE THREAD

**Issue by wangkechun** [bytedance/sonic#6]
```
func TestGoroutineStackExceeds(t *testing.T) {
	var v interface{}
	err := Unmarshal([]byte(`{"a":`+strings.Repeat(`[`, 3000000)+strings.Repeat(`]`, 3000000)+`}`), &v)
	require.Nil(t, err)
}

```

```
=== RUN   TestGoroutineStackExceeds
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0xc02138c3a8 stack=[0xc02138c000, 0xc04138c000]
fatal error: stack overflow

runtime stack:
runtime.throw(0x142f564, 0xe)
	/usr/local/Cellar/go/1.15.2/libexec/src/runtime/panic.go:1116 +0x72
runtime.newstack()
	/usr/local/Cellar/go/1.15.2/libexec/src/runtime/stack.go:1060 +0x78d
runtime.morestack()
	/usr/local/Cellar/go/1.15.2/libexec/src/runtime/asm_amd64.s:449 +0x8f

goroutine 19 [running]:
runtime.mallocgc(0x100, 0x13bf520, 0x1, 0xaaaaaaaaaaaaaaaa)
	/usr/local/Cellar/go/1.15.2/libexec/src/runtime/malloc.go:903 +0xa8b fp=0xc02138c3b8 sp=0xc02138c3b0 pc=0x100fc8b
runtime.makeslice(0x13bf520, 0x0, 0x10, 0x0)
	/usr/local/Cellar/go/1.15.2/libexec/src/runtime/slice.go:98 +0x6c fp=0xc02138c3e8 sp=0xc02138c3b8 pc=0x104f82c
github.com/bytedance/sonic/decoder.decodeArray(0x100)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:134 +0xc6 fp=0xc02138c488 sp=0xc02138c3e8 pc=0x125f606
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138c528 sp=0xc02138c488 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a665300)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138c5c8 sp=0xc02138c528 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138c668 sp=0xc02138c5c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a665200)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138c708 sp=0xc02138c668 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138c7a8 sp=0xc02138c708 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a665100)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138c848 sp=0xc02138c7a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138c8e8 sp=0xc02138c848 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a665000)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138c988 sp=0xc02138c8e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138ca28 sp=0xc02138c988 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664f00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138cac8 sp=0xc02138ca28 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138cb68 sp=0xc02138cac8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664e00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138cc08 sp=0xc02138cb68 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138cca8 sp=0xc02138cc08 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664d00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138cd48 sp=0xc02138cca8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138cde8 sp=0xc02138cd48 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664c00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138ce88 sp=0xc02138cde8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138cf28 sp=0xc02138ce88 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664b00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138cfc8 sp=0xc02138cf28 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d068 sp=0xc02138cfc8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664a00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d108 sp=0xc02138d068 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d1a8 sp=0xc02138d108 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664900)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d248 sp=0xc02138d1a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d2e8 sp=0xc02138d248 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664800)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d388 sp=0xc02138d2e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d428 sp=0xc02138d388 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664700)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d4c8 sp=0xc02138d428 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d568 sp=0xc02138d4c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664600)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d608 sp=0xc02138d568 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d6a8 sp=0xc02138d608 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664500)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d748 sp=0xc02138d6a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d7e8 sp=0xc02138d748 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664400)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d888 sp=0xc02138d7e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138d928 sp=0xc02138d888 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664300)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138d9c8 sp=0xc02138d928 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138da68 sp=0xc02138d9c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664200)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138db08 sp=0xc02138da68 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138dba8 sp=0xc02138db08 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664100)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138dc48 sp=0xc02138dba8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138dce8 sp=0xc02138dc48 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a664000)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138dd88 sp=0xc02138dce8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138de28 sp=0xc02138dd88 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663f00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138dec8 sp=0xc02138de28 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138df68 sp=0xc02138dec8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663e00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e008 sp=0xc02138df68 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e0a8 sp=0xc02138e008 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663d00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e148 sp=0xc02138e0a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e1e8 sp=0xc02138e148 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663c00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e288 sp=0xc02138e1e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e328 sp=0xc02138e288 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663b00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e3c8 sp=0xc02138e328 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e468 sp=0xc02138e3c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663a00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e508 sp=0xc02138e468 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e5a8 sp=0xc02138e508 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663900)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e648 sp=0xc02138e5a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e6e8 sp=0xc02138e648 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663800)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e788 sp=0xc02138e6e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e828 sp=0xc02138e788 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663700)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138e8c8 sp=0xc02138e828 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138e968 sp=0xc02138e8c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663600)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138ea08 sp=0xc02138e968 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138eaa8 sp=0xc02138ea08 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663500)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138eb48 sp=0xc02138eaa8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138ebe8 sp=0xc02138eb48 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663400)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138ec88 sp=0xc02138ebe8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138ed28 sp=0xc02138ec88 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663300)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138edc8 sp=0xc02138ed28 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138ee68 sp=0xc02138edc8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663200)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138ef08 sp=0xc02138ee68 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138efa8 sp=0xc02138ef08 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663100)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f048 sp=0xc02138efa8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f0e8 sp=0xc02138f048 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a663000)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f188 sp=0xc02138f0e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f228 sp=0xc02138f188 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662f00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f2c8 sp=0xc02138f228 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f368 sp=0xc02138f2c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662e00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f408 sp=0xc02138f368 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f4a8 sp=0xc02138f408 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662d00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f548 sp=0xc02138f4a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f5e8 sp=0xc02138f548 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662c00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f688 sp=0xc02138f5e8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f728 sp=0xc02138f688 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662b00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f7c8 sp=0xc02138f728 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f868 sp=0xc02138f7c8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662a00)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138f908 sp=0xc02138f868 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138f9a8 sp=0xc02138f908 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662900)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138fa48 sp=0xc02138f9a8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138fae8 sp=0xc02138fa48 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662800)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138fb88 sp=0xc02138fae8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138fc28 sp=0xc02138fb88 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662700)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138fcc8 sp=0xc02138fc28 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138fd68 sp=0xc02138fcc8 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662600)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138fe08 sp=0xc02138fd68 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138fea8 sp=0xc02138fe08 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662500)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc02138ff48 sp=0xc02138fea8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc02138ffe8 sp=0xc02138ff48 pc=0x17f50a2
github.com/bytedance/sonic/decoder.decodeArray(0xc01a662400)
	/Users/wkc/Downloads/sonic/decoder/generic_amd64.s:147 +0xfa fp=0xc021390088 sp=0xc02138ffe8 pc=0x125f63a
__decode_value_17f5000()
	?:0 +0xa2 fp=0xc021390128 sp=0xc021390088 pc=0x17f50a2
...additional frames elided...
created by testing.(*T).Run
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1178 +0x386

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000082a80, 0x1435bea, 0x19, 0x144b578, 0x108b901)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1179 +0x3ad
testing.runTests.func1(0xc000082900)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1449 +0x78
testing.tRunner(0xc000082900, 0xc0000d5de0)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1127 +0xef
testing.runTests(0xc0001cfaa0, 0x16918e0, 0x87, 0x87, 0xc02538cc519f3ee8, 0x8bb2e696ee, 0x16d0860, 0x100f570)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1447 +0x2e8
testing.(*M).Run(0xc0000ec600, 0x0)
	/usr/local/Cellar/go/1.15.2/libexec/src/testing/testing.go:1357 +0x245
main.main()
	_testmain.go:323 +0x138
exit status 2
FAIL	github.com/bytedance/sonic	1.647s
```

**Comment by chenzhuoyu** [comment]
Known issue, we are working on it.

**Comment by chenzhuoyu** [comment]
@wangkechun Fixed in #29.
